pkg/cmd/server/admin/create_signercert.go

```diff
@@ -99,9 +96,9 @@ func (o CreateSignerCertOptions) CreateSignerCert() (*crypto.CA, error) {
 		ca, written, err = crypto.EnsureCA(o.CertFile, o.KeyFile, o.SerialFile, o.Name)
 	}
 	if written {
-		fmt.Fprintf(o.Output.Get(), "Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
+		glog.V(3).Infof("Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
```

Comment:
> On Jun 5, 2015, at 10:08 AM, Luke Meyer <notifications@github.com> wrote:
> 
> In pkg/cmd/server/admin/create_signercert.go:
> 
> > @@ -99,9 +96,9 @@ func (o CreateSignerCertOptions) CreateSignerCert() (*crypto.CA, error) {
> >  		ca, written, err = crypto.EnsureCA(o.CertFile, o.KeyFile, o.SerialFile, o.Name)
> >  	}
> >  	if written {
> > -		fmt.Fprintf(o.Output.Get(), "Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
> > +		glog.V(3).Infof("Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
> If your code needs to not have this output, pass in a writer that discards the output.
> 
> You basically have two options. Pass down something so that the local routine dispatches the output correctly. Or have the local routine pass up the information to something that has the context for whether to output it. It's kinda ugly either way, but I think the existing way is cleanest. You have to pass things down regardless. Aggregating return values up the stack is yecch.
> 
It's not clean, because start and "create" are not the same type of command.  If you want to react specifically to each creation accept a function that is a callback (see pkg/config/cmd/cmd.go)
> —
> Reply to this email directly or view it on GitHub.
>